packages/hash/frontend/src/pages/[account-slug]/types/entity-type/delete-data-type-modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { faClose } from "@fortawesome/free-solid-svg-icons";
+import {
+  Button,
+  FontAwesomeIcon,
+  IconButton,
+} from "@hashintel/hash-design-system";
+import { Box, Stack, Typography } from "@mui/material";
+import { bindDialog, PopupState } from "material-ui-popup-state/hooks";
+import { ReactNode } from "react";
+import { Modal } from "../../../../components/Modals/Modal";
+
+export interface DeleteDataTypeModalProps {
+  popupState: PopupState;
+  onClose?: () => void;
+  onDelete?: () => void;
+  dataTypeCount?: number;
+  arrayCount?: number;
+  propertyObjectCount?: number;
```

Comment:
These properties aren't actually optional 

```suggestion
  popupState: PopupState;
  onClose: () => void;
  onDelete?: () => void;
  dataTypeCount: number;
  arrayCount: number;
  propertyObjectCount: number;
```